cdap-app-templates/cdap-etl/cdap-etl-tools/src/main/java/co/cask/cdap/etl/tool/UpgradeTool.java

```diff
@@ -0,0 +1,284 @@
+      AppRequest<ETLBatchConfig> updateRequest = new AppRequest<>(batchArtifact, newConfig);
+      appClient.update(appId, updateRequest);
+    } else {
+      OldETLRealtimeConfig oldConfig = GSON.fromJson(appDetail.getConfiguration(), OldETLRealtimeConfig.class);
+      ETLRealtimeConfig newConfig = oldConfig.getNewConfig();
+      AppRequest<ETLRealtimeConfig> updateRequest = new AppRequest<>(realtimeArtifact, newConfig);
+      appClient.update(appId, updateRequest);
+    }
+  }
+
+  public static void main(String[] args) throws Exception {
+
+    Options options = new Options()
+      .addOption(new Option("h", "help", false, "Print this usage message."))
+      .addOption(new Option("u", "uri", true, "CDAP instance URI to interact with in the format " +
+        "[http[s]://]<hostname>:<port>. Defaults to localhost:10000."))
+      .addOption(new Option("a", "accesstoken", true, "File containing the access token to use when interacting " +
+        "with a secure CDAP instance."))
+      .addOption(new Option("t", "timeout", true, "Timeout in milliseconds to use when interacting with the " +
+        "CDAP RESTful APIs. Defaults to " + DEFAULT_READ_TIMEOUT + "."))
+      .addOption(new Option("n", "namespace", true, "Namespace to perform the upgrade in. If none is given, " +
+        "pipelines in all namespaces will be upgraded."))
+      .addOption(new Option("p", "pipeline", true, "Name of the pipeline to upgrade. If specified, a namespace " +
+        "must also be given."))
+      .addOption(new Option("f", "configfile", true, "File containing old application details to update. " +
+        "The file contents are expected to be in the same format as the request body for creating an " +
+        "ETL application from one of the etl artifacts. " +
+        "It is expected to be a JSON Object containing 'artifact' and 'config' fields." +
+        "The value for 'artifact' must be a JSON Object that specifies the artifact scope, name, and version. " +
+        "The value for 'config' must be a JSON Object specifies the source, transforms, and sinks of the pipeline, " +
+        "as expected by older versions of the etl artifacts."));
+
+    CommandLineParser parser = new BasicParser();
+    CommandLine commandLine = parser.parse(options, args);
+
+    if (commandLine.hasOption("h")) {
+      HelpFormatter helpFormatter = new HelpFormatter();
+      helpFormatter.printHelp(UpgradeTool.class.getName(), "Upgrades Hydrator pipelines created for 3.2.x versions" +
+        "of the cdap-etl-batch and cdap-etl-realtime artifacts into pipelines compatible with 3.3.x versions of " +
+        "cdap-etl-batch and cdap-etl-realtime. Connects to an instance of CDAP to find any 3.2.x pipelines, then " +
+        "upgrades those pipelines.", options, "");
+      System.exit(0);
+    }
+
+    if (commandLine.hasOption("f")) {
+      convertFile(commandLine.getOptionValue("f"));
+      System.exit(0);
```

Comment:
so we exit after convert file ? but that just prints the new config right? is that the expected behavior ?